### ISSUE TITLE
Domain level SSL Option buttons can break webmin and virtualmin and need a warning box

### ISSUE THREAD

**Issue by durangod** [webmin/webmin#431]
Hi, the buttons on the bottom of this page (see pic also attached)

virtualmin ->server configuration ->manage ssl certificate -> current certificate

really need a warning on them, if you install a domain certificate thinking that it will just affect oru domain version of webmin and nothing else, and you press that first button and your cert is not for your main server hostname, you will break webmin and virtualmin and you will have to edit the httpd conf file and remove all the wrong links and redo all your cert files from the start. 

I know what it reads now as i have read it several times now, it is a nice feature to have if your doing your first virt server of the same host and domain and everything matches or you have a wildcard cert for everything.  

However if you are not and you dont read that carefully you will think it only affects your domain and it does not.  It overwrites the main data for the main host just as it says it does.  

I not sure what the actuall solution is for stupid people like me that dont read that carefully and just assume what it means.  

I know why its there, but maybe it should not be so powerful and not allow a domain to edit the main host data.     Maybe even put it in another category so that people will intentionally have to access it to use it or behind another button.  Or better yet at least maybe put a red box around it with some text that says "WARNING"  which will get someones attention to make sure they read it.   

And the other buttons there as well, if im running postfix or dovecot source from my main host  then if i have a different domain name than the host and i hit those buttons, i think it will break those too.  I know its a convenient feature if all the certs cover everything but what if they dont. 

Just something added to warn people or isolate the affect to just the domain.  Main Host options really should not be on the virtual host level should they, im not sure. 

I dont know if dovecot or postfix can be run on a domain level, i think it would be a pain. So i think maybe the simplist thing is to just put a big fat red box around that section with warning text as it can save people from going thru what i just went thru trying to get my webmin and virtmin back online. 

See Pic... 
Thanks again and thanks for your hard work, it is appreciated so much :)

![domainssloptions](https://cloud.githubusercontent.com/assets/6512600/18772685/49601a68-8106-11e6-8280-d6b54de27bfc.jpg)

**Comment by jcameron** [comment]
I'm a bit confused here ... but are you referring to the "Copy to Webmin" button that causes this problem? At worst that could cause people to see the wrong cert when accessing the Virtualmin/Webmin port - but it shouldn't completely break the UI.

**Comment by durangod** [comment]
Yes that button... when i pressed that button the whole inside of the UI went away, the template frame was still there but the inside data section was gone.  And i had to go into the configs and change all the paths back and reload all my certs again for both. So all i know is that i had the cert for the host fully functional at name.domain.com and had it working for a few weeks so it was ok.  And then i was in virtmin and loaded my domain cert and pressed that button and it all got messed up. 

I do have my dns via cloudflare free plan if that makes a difference. Its up to you of course, if you dont feel it needs a warning message there.  

An other note, you do a great job and i honestly dont know how you keep all this information in your head without it exploding.  I thought i knew quite a bit until i set up my own dedicated server then i realized quickly just how little i know  :)  Apache and other documentation in this field your in is like trying to follow lightning, jumping all over the place just to get one little answer.   I totally appreciate what you do and what you have done.

**Comment by durangod** [comment]
Just to expand and give you some update here as this just happened with my webmin too so now i understand why. 

The first thing i installed was webmin and at the time i had no idea that i might need anything else.  Later i installed virtualmin and usermin which both run under  my main hostname (the same as my webmin)  

So for example my host name is host.name.com and webmin runs under that(as root).  When i installed virtualmin it also runs under the same name in the browser and the same port as webmin(as root).  So i have them both on the menu i just change back and forth.  So they run under host.name.com:port

My virtual server webmin (that belongs to the domain admin) also runs under the same host.domain.com i just sign into it with a different name. (i dont use it much so i really dont need it)

When i added usermin it is the same way, even though it only has menu access to newdomain.com  i still sign into it with host.name.com:port (but different port number than webmen) 

So all of them use the same cert.  The only thing that does not use the same cert is the actual domain cert which is just for the domain.  

So when i submitted that domain cert to webmin the system was probably thinking that i wanted to copy it to my domain webmin and then today my domain usermin.  But since both of them are useing the host port setup and are actually signed into under a dif name all together then thats why it messed up.  

The software did what it was told to do and it did so.  What probably needs to happen is maybe a check to see if the domains match if not then a warning or a comfirm of some kind, or even just a text warning. 

What happening is that a domain cert is being uploaded and overwriting a hostname cert (they dont match one is x.com and one is z.com).  And so the host name connection no longer works because the name is different.   

The intention of the upload to webmin button is to upload it to the users webmin not the main controllers host webmin.   But there is no check for that, procedural logic to tell the system the difference between the host webmin and the users webmin.  

I hope that makes sense.  Is there a way to keep that option inside a boundry so that it does not affect a higher level above itself and will just do the users webmin, and leave the host webmin alone? 

Food for thought... :)
Dave

**Comment by durangod** [comment]
here is a quick flow to help understand... :)

![webminflow](https://cloud.githubusercontent.com/assets/6512600/18811262/a9c665b2-8267-11e6-9dd0-2c9b000bfef5.jpg)

**Comment by brimansoftware** [comment]
durangod,
can I ask what you did to resolve this problem?

I just got bit by this also and unfortunately did not read this post until after damage was done :(

**Comment by durangod** [comment]
Hi, i didnt, i decided to reformat, start over, and go with cpanel.

**Comment by brimansoftware** [comment]
Ok thanks for answering.

I guess I will keep researching 

I agree with you the actions are misleading as I thought it would apply to the individual domain not the host name of the server.
:/

**Comment by qooob** [comment]
@durangod Not sure how _cPanel_ would solve it. It has no power of _**Webmin/Virtualmin**_ combined. Everything is pretty _hardcoded_, forcing a user to manage the server the certain, I would say, far not flexible way.

Besides, it costs much more?

What do you think we could improve to become better than _cPanel_? What makes you favor to _cPanel_?

**Comment by brimansoftware** [comment]
Well I got everything back on track. Was enough, I went to the .conf files in /webmin and /usermin and changed the name of the certificate used back to the .pem cert that was created during webmin/usermin install. Then deleted the website certs that I made webmin/usermin by mistake.

Everything seems to be fine now

**Comment by ardabeyazoglu** [comment]
I just had the same problem with current release. It took my 2 hours to try different solutions, and the best solution seems to create just another virtual server with exact same domain name as webmin's. Then issuing a letsencrypt ssl for this domain, will be used for webmin admin panel.

**Comment by chris001** [comment]
It's 2020 now. By default, real official LetsEncrypt TLS certs, with properly matching domain names, on the main webmin domain, and on all virtual servers, should be made to be the default initial configuration. The self-signed certs should be made to be the secondary, non-default option, which almost nobody would ever end up choosing...

**Comment by ardabeyazoglu** [comment]
@chris001 I totally agree, but the development is not that active appearantly here. However, I still think its a better alternative to cpanel etc. I enabled default letsencrypt from the settings and get rid of self-signed certificates.

**Comment by iliarostovtsev** [comment]
> .. Let's Encrypt SSL on the main Webmin domain and on all virtual servers, should be made to be the default initial configuration.

How would that be possible? Webmin cannot know neither what is main nor default domain for you

For virtual servers, it's better to use Virtualmin module.

**Comment by chris001** [comment]
@iliaross 
You're right.  What I meant is, any domain that  gets added to webmin, for example `mydomain.com`, webmin should interpret this domain, by default, as wanting a LetsEncrypt cert, and not a self signed cert, for all domain-associated web services: apache/nginx web, webmin/usermin/filemin web, IMAP, SMTP, FTP, etc, and to install LetsEncrypt certs on those web services, and don't generate or attempt to install any self signed certs, by default.

**Comment by iliarostovtsev** [comment]
> What I meant is, any domain that gets added to webmin, for example mydomain.com, webmin should interpret this domain, by default, as wanting a LetsEncrypt cert,

Okay, Chris, now it makes sense. Sounds like an improvement.

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.
